include/swift/ABI/MetadataValues.h

```diff
@@ -606,6 +615,55 @@ class TargetFunctionTypeFlags {
 };
 using FunctionTypeFlags = TargetFunctionTypeFlags<size_t>;
 
+template <typename int_type> class TargetParameterTypeFlags {
+  enum : int_type {
+    InOutMask    = 0x10000000U,
+    SharedMask   = 0x01000000U,
+    VariadicMask = 0x00100000U,
+  };
```

Comment:
I've just looked at some other places, and I didn't know that it's more expensive, will fix, thanks!

Reply:
It's pretty minor, but yes, it is often the case that you can do something with a very small integer in one instruction that takes multiple instructions with a larger integer, so if it's otherwise an arbitrary choice, prefer to use smaller integers for more important flags.

Reply:
Understood, thanks!